### PR TITLE
Apply Azure AD, B2C changes on next to fix provider semantics

### PR DESCRIPTION
**Pull request by BenjaminWFox**
**What**: 
1. Created new provider `azure-ad` containing what was previously `azure-ad-b2c`

2. Updated content for provider `azure-ad-b2c` to reflect correct configuration for that provider

**Why**:

https://github.com/nextauthjs/next-auth/issues/1586

**How**:

Changes to Provider JS & MD files, and updating the provider index.

**Checklist**:

- [x] Documentation
- [ ] Tests (May be N/A)
- [ ] Ready to be merged
  - Should be reviewed for correctness
  - Do I need to do something specific to generate types?
  - Doesn't pass deployment, unsure if that's expected at this point on `next`
      

# * This will probably be a breaking change *
Anyone who previously has implemented what is currently provider `azure-ad-b2c`/`Providers.AzureADB2C` will need to update their provider to be `azure-ad`/`Providers.AzureAD`. There should be no other changes needed.

### CODE REVIEW

**Comment by balazsorban44** [comment]
Not sure why the Vercel deployment fails, I switched to your branch locally, and the build was successful. 🤷

File: src/providers/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  Reddit,
+  Salesforce,
+  Slack,
+  Spotify,
+  Strava,
+  Twitch,
+  Twitter,
+  VK,
+  Yandex
+}
```
Comment (BenjaminWFox) [inline src/providers/index.js:81]: @balazsorban44 Fixing merge conflicts I saw this file was removed on the `next` branch but it is still referenced in the dev app config file [https://github.com/nextauthjs/next-auth/blob/next/app/pages/api/auth/%5B...nextauth%5D.js](https://github.com/nextauthjs/next-auth/blob/next/app/pages/api/auth/%5B...nextauth%5D.js)

Is the removal of the providers `index.js` intentional?

**Comment by BenjaminWFox** [comment]
@balazsorban44 also is this Types lint error a known issue? Unsure if I should do something in my code to fix that or if it will be fixed elsewhere on `next`.

```
Error: .../next-auth/types/adapters.d.ts:2:49
ERROR: 2:49  expect  TypeScript@4.3 compile error:
Cannot find module 'typeorm' or its corresponding type declarations.

.../next-auth/types/index.d.ts:5:35
ERROR: 5:35  expect  TypeScript@4.3 compile error:
Cannot find module 'typeorm' or its corresponding type declarations.
```

File: src/providers/index.js (hunk 1 above)
Comment (balazsorban44) [reply src/providers/index.js:81]: Yes, it is autogenerated now! See #1782

**Comment by balazsorban44** [comment]
Yes, I guess it is "expected"... I am experimenting with cutting the `typeorm` package out of the core (making it optional to those actually use it because takes up around 70% of the bundle...), but we still reference stuff in types from it. I'll have a look next time I am working on the next branch!

File: src/providers/azure-ad-b2c.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
-export default function AzureADB2C(options) {
-  const tenant = options.tenantId ? options.tenantId : "common"
+export default (options) => {
```
Comment (balazsorban44) [inline src/providers/azure-ad-b2c.js:1]: Could you make this a named default export function again? Goes together with auto-generating the index.js file

File: src/providers/azure-ad.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+export default (options) => {
```
Comment (balazsorban44) [inline src/providers/azure-ad.js:1]: Same as above, must be a named default export now :slightly_smiling_face:

File: src/providers/azure-ad.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const tenant = options.tenantId ? options.tenantId : 'common'
```
Comment (balazsorban44) [inline src/providers/azure-ad.js:2]: How about:
```suggestion
  const tenant = options.tenantId ?? 'common'
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: src/providers/azure-ad.js (hunk 4 above)
Comment (BenjaminWFox) [reply src/providers/azure-ad.js]: I had no idea that 'suggested change' functionality existed 🤯

Comment (balazsorban44) [reply src/providers/azure-ad.js]: you can even batch my suggestions into a single commit and apply them 😉

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
NOTE to other maintainers: When merging, we should to point out the breaking changes here, following the https://www.conventionalcommits.org/en/v1.0.0/ `BREAKING CHANGE` format for the commit message.

File: www/docs/providers/azure-ad-b2c.md [hunk 5]
```diff
@@ -5,11 +5,7 @@ title: Azure Active Directory B2C
 
 ## Documentation
 
-https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow
-
-## Configuration
-
-https://docs.microsoft.com/en-us/azure/active-directory-b2c/tutorial-create-tenant
+<!-- https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow -->
```
Comment (balazsorban44) [inline www/docs/providers/azure-ad-b2c.md:8]: Why was this commented out? Should we remove the whole `Documentation` section if it is empty or have a better link?

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Hi, @BenjaminWFox! Sorry for being slow on this, but I would like to pick this up now and get this merged! I had a simple comment, could you address that? If everything is fine, then I'll merge this ASAP, to get things ready for `next-auth@4`... :eyes:

**Pull request merged**